test/pages/uat_actions/generate_test_run.spec.js

```diff
@@ -0,0 +1,186 @@
+import lighthouse from '@/modules/lighthouse_service'
+import Alert from '@/components/Alert'
+import flushPromises from 'flush-promises'
+
+jest.mock('@/modules/lighthouse_service')
+
+const localVue = createLocalVue()
+localVue.use(BootstrapVue)
+
+describe('UAT Actions', () => {
+  let wrapper, page
+
+  beforeEach(() => {
+    wrapper = mount(GenerateTestRun, {
+      localVue,
+    })
+    page = wrapper.vm
+  })
+
+  it('is a Vue instance', () => {
+    expect(wrapper.findComponent(GenerateTestRun).exists()).toBeTruthy()
+  })
+
+  // components
+  describe('alert', () => {
+    it('has a alert', () => {
+      expect(wrapper.findComponent(Alert).exists()).toBeTruthy()
+    })
+  })
+
+  // data
+  describe('data', () => {
+    it('has default values', () => {
+      expect(page.form).toEqual({
+        numberOfPlates: 1,
+        numberOfPositives: 0,
+      })
+      expect(page.addToDart).toEqual('')
+      expect(page.plateSpecs).toEqual([])
+    })
+  })
+
+  // computed
+  describe('totalPlates', () => {
+    it('totals the number of plates', () => {
+      wrapper = mount(GenerateTestRun, {
+        localVue,
+        data() {
+          return {
+            plateSpecs: [{ numberOfPlates: 2, numberOfPositives: 1 }, { numberOfPlates: 11, numbeOfPpositives: 3 }],
+          }
+        },
+      })
+
+      expect(wrapper.vm.totalPlates).toEqual(2 + 11)
+    })
+  })
+
+  // methods
+  describe('#numberOfPositivesOptions', () => {
+    it('has a default', () => {
+      expect(wrapper.vm.numberOfPositivesOptions()).toHaveLength(97)
+    })
+  })
+
+  describe('#numberOfPlatesOptions', () => {
+    it('has a default', () => {
+      expect(wrapper.vm.numberOfPlatesOptions()).toHaveLength(200)
+    })
+  })
+
+  describe('#add', () => {
+    it('when plateSpecs is an empty list', async () => {
+      wrapper = mount(GenerateTestRun, {
+        localVue,
+        data() {
+          return {
+            form: { numberOfPlates: 1, numberOfPositives: 2 },
+          }
+        },
+      })
+
+      await wrapper.find('#addButton').trigger('click')
+
+      expect(wrapper.vm.plateSpecs).toEqual([{ numberOfPlates: 1, numberOfPositives: 2 }])
+      expect(wrapper.vm.form.numberOfPlates).toEqual(1)
+      expect(wrapper.vm.form.numberOfPositives).toEqual(0)
+    })
+
+    it('when plateSpecs is not empty', async () => {
+      wrapper = mount(GenerateTestRun, {
+        localVue,
+        data() {
+          return {
+            plateSpecs: [{ numberOfPlates: 1, numberOfPositives: 2 }],
+            form: { numberOfPlates: 3, numberOfPositives: 4 },
+          }
+        },
+      })
+
+      await wrapper.find('#addButton').trigger('click')
+
+      expect(wrapper.vm.plateSpecs).toEqual([{ numberOfPlates: 1, numberOfPositives: 2 }, { numberOfPlates: 3, numberOfPositives: 4 }])
+      expect(wrapper.vm.form.numberOfPlates).toEqual(1)
+      expect(wrapper.vm.form.numberOfPositives).toEqual(0)
+    })
+  })
```

Comment:
Worth adding a test that you can add another entry with the same number of positives as an already existing plate spec and that they remain in insertion order.  That's apparently a business requirement.